src/app/auth/auth/auth.component.ts

```diff
@@ -10,69 +15,32 @@ import { AuthService } from "../../services/auth.service";
 })
 export class AuthComponent implements OnInit {
   userForm: FormGroup;
-  formErrors = {
-    email: "",
-    password: ""
-  };
-  validationMessages = {
-    email: {
-      required: "Please enter your email",
-      email: "Please enter your vaild email"
-    },
-    password: {
-      required: "please enter your password",
-      pattern: "Password must contain numbers and letters",
-      minlength: "Please enter more than 4 characters",
-      maxlength: "Please enter less than 25 characters"
-    }
-  };
+  email: FormControl;
+  password: FormControl;
 
-  constructor(
-    private router: Router,
-    private fb: FormBuilder,
-    private auth: AuthService
-  ) {}
+  constructor(private router: Router, private auth: AuthService) {}
 
   ngOnInit() {
-    this.buildForm();
+    this.createFormControls();
+    this.createForm();
   }
 
-  buildForm() {
-    this.userForm = this.fb.group({
-      email: ["", [Validators.required, Validators.email]],
-      password: [
-        "",
-        [
-          Validators.pattern("^(?=.*[0-9])(?=.*[a-zA-Z])([a-zA-Z0-9]+)$"),
-          Validators.minLength(6),
-          Validators.maxLength(25)
-        ]
-      ]
-    });
-
-    this.userForm.valueChanges.subscribe(data => this.onValueChanged(data));
-    this.onValueChanged();
+  createFormControls() {
+    this.email = new FormControl("", [
```

Comment:
you can use `input` `type="email"`. in this case you will not need `pattern("[^ @]*@[^ @]*")`